app/workers/delete_object_hierarchy_worker.rb

```diff
@@ -94,8 +97,28 @@ def destroyable_associations
     object.class.reflect_on_all_associations.select { |reflection| destroyable_association?(reflection) }
   end
 
+  def deletable_associations
+    object.class.reflect_on_all_associations.select { |reflection| deletable_association?(reflection) }
+  end
+
+  class BackgroundDeletionError < StandardError; end
+
   def destroyable_association?(reflection)
-    %i[destroy delete_all].include?(reflection.options[:dependent])
+    should_be_destroyed_or_deleted?(reflection, :background_destroy_associations)
+  end
+
+  def deletable_association?(reflection)
+    should_be_destroyed_or_deleted?(reflection, :background_delete_associations)
+  end
+
+  def should_be_destroyed_or_deleted?(reflection, list_name)
+    list_defined = object.respond_to?(list_name)
```

Comment:
you can use `object.method(list_name)` and rescue from `NameError`

That way you can just do `method.call.include?`

By the way raising an error when it is dependent but not in the list is too hard failure. we should continue the excecution of deleting no?